roles/assemble_template/tasks/main.yml

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Cloudera, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+- name: Discover fragment files
+  ansible.builtin.find:
+    paths: "{{ cluster_template_fragments_directory | mandatory  }}"
+    patterns: "{{ json_regex_pattren | default(omit) }}"
+    use_regex: yes
+    recurse: yes
+  register: fragments
+
+- name: Create a temporary directory
+  ansible.builtin.tempfile:
+    state: directory
+  register: fragments_temp_directory
+
+- name: Loop through fragment files and template them
+  ansible.builtin.template:
+    src: "{{ __fragment.path }}"
+    dest: "{{ fragments_temp_directory.path }}/{{ __fragment.path | basename }}"
+  loop: "{{ fragments.files }}"
+  loop_control: 
+    loop_var: __fragment
+    label: "{{ __fragment.path | basename }}"
+  when: fragments.matched > 0
+
+- name: Create cluster template
+  cloudera.cluster.assemble_cluster_template:
+    src: "{{ fragments_temp_directory.path }}"
+    dest: "{{ fragments_temp_directory.path }}/cluster-template.json"
+
+- name: Set fact for cluster template path
+  ansible.builtin.set_fact:
+    cluster_template_path: "{{ fragments_temp_directory.path }}/cluster-template.json"
```

Comment:
I suggest we add a mandatory default variable for the role for this path -- that would allow one to assemble the file into a known/predetermined location (effectively removing this play at line 45 and updating the `dest` parameter at line 43) and allowing the role to then clean up the templated temp file fragments before completing.

Reply:
Added